src/hu/elte/txtuml/stdlib/math/BigDecimal.java

```diff
@@ -0,0 +1,1308 @@
+	 * Returns a {@code BigDecimal} whose value is the integer part
+	 * of the quotient {@code (this / divisor)} rounded down.  The
+	 * preferred scale of the result is {@code (this.scale() -
+	 * divisor.scale())}.
+	 *
+	 * @param  divisor value by which this {@code BigDecimal} is to be divided.
+	 * @return The integer part of {@code this / divisor}.
+	 * @throws ArithmeticException if {@code divisor==0}
+	 * @since  1.5
+	 */
+	public BigDecimal divideToIntegralValue(BigDecimal divisor);
+
+	/**
+	 * Returns a {@code BigDecimal} whose value is the integer part
+	 * of {@code (this / divisor)}.  Since the integer part of the
+	 * exact quotient does not depend on the rounding mode, the
+	 * rounding mode does not affect the values returned by this
+	 * method.  The preferred scale of the result is
+	 * {@code (this.scale() - divisor.scale())}.  An
+	 * {@code ArithmeticException} is thrown if the integer part of
+	 * the exact quotient needs more than {@code mc.precision}
+	 * digits.
+	 *
+	 * @param  divisor value by which this {@code BigDecimal} is to be divided.
+	 * @param  mc the context to use.
+	 * @return The integer part of {@code this / divisor}.
+	 * @throws ArithmeticException if {@code divisor==0}
+	 * @throws ArithmeticException if {@code mc.precision} {@literal >} 0 and the result
+	 *         requires a precision of more than {@code mc.precision} digits.
+	 * @since  1.5
+	 * @author Joseph D. Darcy
+	 */
+	public BigDecimal divideToIntegralValue(BigDecimal divisor, MathContext mc);
+
+	/**
+	 * Converts this {@code BigDecimal} to a {@code double}.
+	 * This conversion is similar to the
+	 * <i>narrowing primitive conversion</i> from {@code double} to
+	 * {@code float} as defined in section 5.1.3 of
+	 * <cite>The Java&trade; Language Specification</cite>:
+	 * if this {@code BigDecimal} has too great a
+	 * magnitude represent as a {@code double}, it will be
+	 * converted to {@link Double#NEGATIVE_INFINITY} or {@link
+	 * Double#POSITIVE_INFINITY} as appropriate.  Note that even when
+	 * the return value is finite, this conversion can lose
+	 * information about the precision of the {@code BigDecimal}
+	 * value.
+	 *
+	 * @return this {@code BigDecimal} converted to a {@code double}.
+	 */
+	public double doubleValue();
+
+	/**
+	 * Compares this {@code BigDecimal} with the specified
+	 * {@code Object} for equality.  Unlike {@link
+	 * #compareTo(BigDecimal) compareTo}, this method considers two
+	 * {@code BigDecimal} objects equal only if they are equal in
+	 * value and scale (thus 2.0 is not equal to 2.00 when compared by
+	 * this method).
+	 *
+	 * @param  x {@code Object} to which this {@code BigDecimal} is
+	 *         to be compared.
+	 * @return {@code true} if and only if the specified {@code Object} is a
+	 *         {@code BigDecimal} whose value and scale are equal to this
+	 *         {@code BigDecimal}'s.
+	 * @see    #compareTo(BigDecimal)
+	 * @see    #hashCode
+	 */
+	public boolean equals(Object x);
```

Comment:
The `obj` parameter should be type `UUID` instead of `Object` because `Object` is not a txtUML class nor an external class.